pkg/api/serialization_test.go

```diff
@@ -169,10 +174,43 @@ func originFuzzer(t *testing.T, seed int64) *fuzz.Fuzzer {
 			}
 		},
 		func(j *template.Template, c fuzz.Continue) {
-			c.Fuzz(&j.ObjectMeta)
-			c.Fuzz(&j.Parameters)
-			// TODO: replace with structured type definition
-			j.Objects = []runtime.Object{}
+			c.FuzzNoCustom(j)
+			j.Objects = nil
+
+			objs := []runtime.Object{
+				&kapi.Pod{},
+				&extensions.Deployment{},
+				&kapi.Service{},
+				&build.BuildConfig{},
+			}
+
+			for _, obj := range objs {
+				c.Fuzz(obj)
+
+				var codec runtime.Codec
+				switch obj.(type) {
+				case *extensions.Deployment:
+					codec = apitesting.TestCodec(kapi.Codecs, extensionsv1beta1.SchemeGroupVersion)
+				case *build.BuildConfig:
+					codec = apitesting.TestCodec(kapi.Codecs, buildv1.SchemeGroupVersion)
+				default:
+					codec = apitesting.TestCodec(kapi.Codecs, v1.SchemeGroupVersion)
+				}
+
+				b, err := runtime.Encode(codec, obj)
+				if err != nil {
+					t.Error(err)
+					return
+				}
+				b = bytes.TrimRight(b, "\n")
+
+				j.Objects = append(j.Objects,
+					&runtime.Unknown{
+						ContentType: "application/json",
+						Raw:         b,
+					},
+				)
+			}
```

Comment:
could also add an unknown type to make sure we can round trip it. something like:

```go
j.Objects = append(j.Objects, &runtime.Unknown{
	ContentType: "application/json",
	Raw:         []byte(`{"kind":"Foo","apiVersion":"mygroup/v1","complex":{"a":true},"list":["item"],"bool":true,"int":1,"string":"hello"}`),
})
```